nomt/src/store/sync.rs

```diff
@@ -53,29 +47,18 @@ impl Sync {
         };
         bitbox_sync.begin_sync(page_cache, merkle_tx, page_diffs);
         beatree_sync.begin_sync(value_tx.batch);
-
-        let rollback_writeout_wd = rollback_writeout_wd_rx
-            .map(|rollback_writeout_wd| rollback_writeout_wd.recv().unwrap());
-
-        let rollback_start_live;
-        let rollback_end_live;
-        let rollback_prune_to_new_start_live;
-        let rollback_prune_to_new_end_live;
-        if let Some(rollback_writeout_wd) = &rollback_writeout_wd {
-            rollback_start_live = rollback_writeout_wd.rollback_start_live;
-            rollback_end_live = rollback_writeout_wd.rollback_end_live;
-            rollback_prune_to_new_start_live = rollback_writeout_wd.prune_to_new_start_live;
-            rollback_prune_to_new_end_live = rollback_writeout_wd.prune_to_new_end_live;
-        } else {
-            rollback_start_live = 0;
-            rollback_end_live = 0;
-            rollback_prune_to_new_start_live = None;
-            rollback_prune_to_new_end_live = None;
+        if let Some(ref mut rollback) = rollback_sync {
+            rollback.begin_sync();
         }
 
         // TODO: comprehensive error handling is coming later.
         bitbox_sync.wait_pre_meta().unwrap();
         let beatree_meta_wd = beatree_sync.wait_pre_meta().unwrap();
+        let (rollback_start_live, rollback_end_live) = if let Some(ref rollback) = rollback_sync {
+            rollback.wait_pre_meta()
+        } else {
+            (0, 0)
+        };
```

Comment:
this chain of `=` is a little bit hard to follow. Not pushing for a change, just another option that is, just for me, more readable

```rust
        let (rollback_start_live, rollback_end_live) = match rollback_sync {
            Some(ref rollback) => rollback.wait_pre_meta(),
            None => (0, 0)
        }; 
```

Reply:
agreed